core/tools.js

```diff
@@ -1627,6 +1627,21 @@
 					yellowgreen: '#9ACD32'
 				},
 
+				_borderStyle: [
+					'none',
+					'hidden',
+					'dotted',
+					'dashed',
+					'solid',
+					'double',
+					'groove',
+					'ridge',
+					'inset',
+					'outset'
+				],
+
+				_widthRegExp: /^(thin|medium|thick|[\+-]?\d+(\.?\d+|)[a-z%]+|0+(\.?0+|))$/,
```

Comment:
Still this regexp has some issues:
![image](https://user-images.githubusercontent.com/1061942/28663475-c26b239a-72bd-11e7-98b3-6952e3786f6d.png)

You may try to group it somehow so it might be easier to match various scenarios (especially with 0).